src/inc/piece/tetris_piece.h

```diff
@@ -26,9 +27,9 @@ class TetrisPiece : public TetrisComponent
 
   bool accepting_input() override;
 
-  std::vector<std::unique_ptr<RenderComponent>>& get_render_components() override;
+  std::vector<std::unique_ptr<Block>>& get_render_components() override;
```

Comment:
Now that the engine is suitably general, I can store render component implementations as the implementation in game components, and access their interfaces directly. This eliminates the need for casting.